paddle/majel/place.h

```diff
@@ -14,33 +14,24 @@ struct CpuPlace {
 };
 
 struct GpuPlace {
-  GpuPlace(int d) : device(d) {}
+  GpuPlace() {}
 
   // needed for variant equality comparison
-  inline bool operator==(const GpuPlace& o) const { return device == o.device; }
+  inline bool operator==(const GpuPlace&) const { return true; }
 
-  inline bool operator!=(const GpuPlace& o) const { return !(*this == o); }
-
-  GpuPlace() : GpuPlace(0) {}
-  int device;
```

Comment:
We will create one ```Context``` for one device, and then set device id to specific ```Context```.  The device id will be handled in Paddle, not in tensor library. It's sure that we can copy data between different GPU cards. Following is a example:

```
    cudaSetDevice(1);
    cudaDeviceEnablePeerAccess(2,flags); //flags=0
    cudaSetDevice(2);
    cudaDeviceEnablePeerAccess(1,flags); //flags=0

    // Allocate some data
    float *gpu1data, *gpu2data;
    cudaSetDevice(1);
    cudaMalloc(&gpu1data, nbytes);
    cudaSetDevice(2);
    cudaMalloc(&gpu2data, nbytes);
    
    // Do the p2p copy!
    cudaMemcpy(gpu1data, gpu2data, cudaMemcpyDefault);
```

The gpu data block does not hold device id information, but the ```Context``` does.